explorer/explorer_macos.go

```diff
@@ -0,0 +1,110 @@
+// SPDX-License-Identifier: Unlicense OR MIT
+
+//go:build darwin && !ios
+// +build darwin,!ios
+
+package explorer
+
+/*
+#cgo CFLAGS: -Werror -xobjective-c -fmodules -fobjc-arc
+
+@import AppKit;
+
+@interface explorer_macos:NSObject
++ (void) exportFile:(CFTypeRef)viewRef name:(char*)name id:(int32_t)id;
++ (void) importFile:(CFTypeRef)viewRef ext:(char*)ext id:(int32_t)id;
+@end
+
+static void exportFile(CFTypeRef viewRef, char * name, int32_t id) {
+	[explorer_macos exportFile:viewRef name:name id:id];
+}
+
+static void importFile(CFTypeRef viewRef, char * ext, int32_t id) {
+	[explorer_macos importFile:viewRef ext:ext id:id];
+}
```

Comment:
I don't understand the rationale for defining an interface for this. Why not simply move the implementation of `explorer_macos`'s methods into the static functions? Is there an objective-C limitation here that I'm missing?

Reply:
It have a reason....

If I remember correctly, that is because it requires a callback and that function doesn't exists on `import "C"` for some reason. I hit some error when I try to call `importCallback` function, directly from `import "C"` code. 

The workaround was to call it on `.m`. So, the `.m` calls `exportCallback` and `importCallback` (those functions are declared on golang-side, using `//export`). Those exported functions are imported on `.m` using `#include "_cgo_export.h"` (it not possible to use that `#include` on `import "C"` too).

So basically, the `.m` implements the `explorer_macos` importing `_cgo_export.h` (which contains the exported golang function). Then, the `import "C"` code will use the implementation defined on `.m`.

Reply:
If you try to move the functions from `.m` to  inside `static void ....` you will hit the same compile error, I guess, saying that the callback functions isn't defined. I'm almost sure that was the reason for that.

Reply:
Could we then simplify by defining everything in the external `.m` file? That way we have access to symbols defined in the Go side by importing the CGO header, and our implementation isn't split across two files. I think we'd need to declare the function prototypes for the C functions in the Go file, but that would be all of the C code in the `.go` files.

Reply:
I never tried it, but maybe using `extern void exportFile(CFTypeRef viewRef, char * name, int32_t id)` on `import "C"` and define it on `.m` may work. _That was my first time playing with Objective-C interfaces (which is mandatory for iOS), so..._

I didn't tried to combine everyhing on `.m` because I thought that `.m` is just used for implementing methods, that is why I create another `@interface`. The `@interface` definition can also be moved to another header file, `explorer_macos.h` (but in this case, it will not be required, but is required for iOS anyway).

Reply:
Objective-C is _fun_, huh? :P I'm not even sure you'll need to use `extern` to declare those functions, though I could be mistaken.

> I thought that .m is just used for implementing methods

I thought `.m` files in Objective-C were the equivalent of `.c` files in C. What I mean by that is that the implementation of everything goes in there, not just methods.

My overall goal here is to simplify the native code portion of the integration as much as possible. Knowledge of Objective-C is scarce (and dying out), so we should try to minimize the amount of it that we need to maintain.

Reply:
I fix that on tha latest commit (`explorer: [macos] move functions to .m file`).